message/message.go

```diff
@@ -349,6 +396,19 @@ func (gsr GraphSyncRequest) Selector() []byte { return gsr.selector }
 // Priority returns the priority of this request
 func (gsr GraphSyncRequest) Priority() GraphSyncPriority { return gsr.priority }
 
+// Extension returns the content for an extension on a response, or errors
+// if extension is not present
+func (gsr GraphSyncRequest) Extension(name GraphSyncExtensionName) ([]byte, error) {
+	if gsr.extensions == nil {
+		return nil, ErrExtensionNotPresent
+	}
+	val, ok := gsr.extensions[string(name)]
+	if !ok {
+		return nil, ErrExtensionNotPresent
+	}
+	return val, nil
```

Comment:
Consider returning `([]byte, bool)`, leaving the interpretation of whether the missing value is an "error" up to the caller.